### PR TITLE
[25.0 Release] Hotfixes for demo

### PR DESCRIPTION
**Pull request by ridz1208**
## Brief summary of changes

1. RESET FOREIGN KEY CHECKS TO 1 !!!IMPORTANT!!! @laemtl please review asap
2. Fixes 9999 drop tables, adds omitted tables in appropriate order
3. adds editor config setting for YML files
4. recreate release patch in correct chronological order
5. restore publication patch accidentally added to cleanups (publication module should be re-tested)
6. remove instrument permissions from config.xml for RB
7. remove corrupting old subproject files from RB
8. update migration.md

### CODE REVIEW

**Comment by driusan** [comment]
PHPCS is failing

```
FILE: ...e/runner/work/Loris/Loris/tools/exporters/DB_dump_table_data.php
----------------------------------------------------------------------
FOUND 6 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 97 | ERROR | [x] Spaces must be used to indent lines; tabs are not
    |       |     allowed
 97 | ERROR | [x] Multi-line function call not indented correctly;
    |       |     expected 8 spaces but found 1
 97 | ERROR | [x] Line indented incorrectly; expected at least 4
    |       |     spaces, found 1
 98 | ERROR | [x] Spaces must be used to indent lines; tabs are not
    |       |     allowed
 98 | ERROR | [x] Multi-line function call not indented correctly;
    |       |     expected 8 spaces but found 1
 98 | ERROR | [x] Line indented incorrectly; expected at least 4
    |       |     spaces, found 1
----------------------------------------------------------------------
PHPCBF CAN FIX THE 6 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

File: SQL/Release_patches/24.1_To_25.0_upgrade.sql [hunk 1]
```diff
@@ -498,7 +451,8 @@ INSERT INTO ConfigSettings
     Name="gui";
 
 INSERT INTO Config (ConfigID, Value) SELECT ID, 'false' FROM ConfigSettings WHERE Name="useEEGBrowserVisualizationComponents";
-CREATE INDEX `i_violations_resolved_extid_type` ON `violations_resolved` (`ExtID`, `TypeTable`);
+
```
Comment (driusan) [inline SQL/Release_patches/24.1_To_25.0_upgrade.sql:454]: I don't think we should be changing this. Was there a problem with the order that it was, even if it wasn't chronological? That's the order that the patch was in when it was tested.

File: raisinbread/RB_files/RB_ConfigSettings.sql [hunk 2]
```diff
@@ -118,6 +118,6 @@ INSERT INTO `ConfigSettings` (`ID`, `Name`, `Description`, `Visible`, `AllowMult
...
 INSERT INTO `ConfigSettings` (`ID`, `Name`, `Description`, `Visible`, `AllowMultiple`, `DataType`, `Parent`, `Label`, `OrderNumber`) VALUES (129,'UserMaximumDaysInactive','The maximum number of days since last login before making a user inactive.',1,0,'text',1,'Maximum Days Before Making User Inactive',30);
 INSERT INTO `ConfigSettings` (`ID`, `Name`, `Description`, `Visible`, `AllowMultiple`, `DataType`, `Parent`, `Label`, `OrderNumber`) VALUES (130,'DownloadPath','Where files are downloaded',1,0,'text',26,'Downloads',4);
-INSERT INTO `ConfigSettings` (`ID`, `Name`, `Description`, `Visible`, `AllowMultiple`, `DataType`, `Parent`, `Label`, `OrderNumber`) VALUES (131,'EEGUploadIncomingPath', 'Path to the upload directory for incoming EEG studies', 1, 0, 'text', 26, 'EEG Incoming Directory', 15);
+INSERT INTO `ConfigSettings` (`ID`, `Name`, `Description`, `Visible`, `AllowMultiple`, `DataType`, `Parent`, `Label`, `OrderNumber`) VALUES (131,'EEGUploadIncomingPath','Path to the upload directory for incoming EEG studies',1,0,'text',26,'EEG Incoming Directory',15);
```
Comment (driusan) [inline raisinbread/RB_files/RB_ConfigSettings.sql:121]: Can you move all the RB changes to a different PR? I think that can be merged pretty easily.

File: SQL/Archive/25.0/2022-03-03-AddHEDTags.sql [hunk 3]
```diff
@@ -44,6 +44,7 @@ INSERT INTO physiological_event_file (PhysiologicalFileID, FilePath, FileType)
 UPDATE physiological_task_event te
     SET EventFileID=(SELECT EventFileID FROM physiological_event_file WHERE PhysiologicalFileID=te.PhysiologicalFileID)
 ;
+SET FOREIGN_KEY_CHECKS= 1;
```
Comment (driusan) [inline SQL/Archive/25.0/2022-03-03-AddHEDTags.sql:47]: I don't get why this is flagged as "IMPORTANT" in the description. It's a session variable, it'll be re-enabled after the patch is sourced even with this missing.

File: SQL/Release_patches/24.1_To_25.0_upgrade.sql (hunk 1 above)
Comment (ridz1208) [reply SQL/Release_patches/24.1_To_25.0_upgrade.sql:454]: we don't know.... we were disabling foreign keys in the same patch

File: SQL/Archive/25.0/2022-03-03-AddHEDTags.sql (hunk 3 above)
Comment (ridz1208) [reply SQL/Archive/25.0/2022-03-03-AddHEDTags.sql:47]: the patch is the entire release patch because its concatenated into it

Comment (laemtl) [reply SQL/Archive/25.0/2022-03-03-AddHEDTags.sql:47]: I think it is fine, the FOREIGN_KEY_CHECKS is disabled for the same set of statements on both patches.
